main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completion.scala

```diff
@@ -524,6 +534,15 @@ object Completion {
     */
   case class SnippetCompletion(name: String, snippet: String, documentation: String) extends Completion
 
+  /**
+   * Represents a Snippet completion
+   *
+   * @param name          the name of the snippet.
+   * @param textEdit      the TextEdit.
+   * @param documentation a human-readable string that represents a doc-comment.
+   */
+  case class MagicMatchCompletion(name: String, textEdit: TextEdit, documentation: String) extends Completion
```

Comment:
The idea with this case class would be not to store the textEdit but rather to store information enough to compute the text from that information. Otherwise the class does not do much. Makes sense?